src/plainer.ts

```diff
@@ -155,27 +156,40 @@ export const walker = (
   identities: Map<any, any[][]>,
   superJson: SuperJSON,
   path: any[] = [],
-  objectsInThisPath: any[] = []
+  objectsInThisPath: any[] = [],
+  seenObjects = new Map<unknown, Result>()
 ): Result => {
-  if (!isPrimitive(object)) {
+  const primitive = isPrimitive(object);
+
+  if (!primitive) {
     addIdentity(object, path, identities);
+
+    const seen = seenObjects.get(object);
+    if (seen) {
+      // short-circuit result if we've seen this object before
+      return seen;
+    }
```

Comment:
@Skn0tt:
would it be OK to return `null` here under some sort of options flag? it would make big outputs a lot smaller

Reply:
I like that idea conceptually - less duplicated data is always better - but I agree it's both a breaking change (the output JSON wouldn't have the same structure anymore), and it makes SuperJSON output harder to understand.

I'm assuming you'd like to reduce the output size to save bandwidth, right? Maybe we could perform some kind of benchmark on the difference this would make in practice - I could imagine that with Gzip / Brotli, the difference isn't that big. If we find that the difference is significant even with compression, then I agree we should add a flag for this behaviour! (not in this PR though)

Reply:
Will do a follow-up PR! 👀